### DISCUSSION TITLE
Unable to build wasm in VS 2022

### DISCUSSION THREAD

**Discussion by Robotuner** [unoplatform/uno#8776] [Q&A] [answered]
My Uno project can build UWP and WASM in VS 2019, but when I switch to VS 2022, the UWP project builds but WASM give me a VS Error:
One or more errors occurred.  Failed to launch debug adapter.  
Cannot connect to runtime process, timeout after 10000 ms - (reason: Cannot connect to the target: connect ECONNREFUSED 127.0.0.1:17032)
The output window reports:
Output from build:
Build started...
========== Build: 0 succeeded, 0 failed, 1 up-to-date, 0 skipped ==========
========== Deploy: 0 succeeded, 0 failed, 0 skipped ==========
Output from Deployment:
IsEmulatorSupported: START
	Is64BitOperatingSystem = True
DetectSlat: START
	isHyperVRunning = True
DetectSlat: END (Success, 0:00:00.069)
	SlatState = SlatAndHWVEnabled
	IsEmulatorSupported = True
IsEmulatorSupported: END (Success, 0:00:00.128)

Has anyone experienced this?  What is the cause?

**Comment by jeromelaban** [comment, marked as answer]
Interesting issue. Can you try running the Wasm app without the debugger?

**Reply by Robotuner** [reply]
Tried a Release build, Still doesn't start.  The build output is slightly different.
Build started...
1>------ Build started: Project: UnoPhotoViewer.Wasm, Configuration: Release Any CPU ------
1>Skipping analyzers to speed up the build. You can execute 'Build' or 'Rebuild' command to run analyzers.
1>Uno.UI is using Roslyn generators
1>C:\Code\Photo\UnoPhotoViewer\UnoPhotoViewer.Shared\Support\NotWasmMediaTemplateSelector.cs(46,30,46,32): warning CS0168: The variable 'ex' is declared but never used
1>UnoPhotoViewer.Wasm -> C:\Code\Photo\UnoPhotoViewer\UnoPhotoViewer.Wasm\bin\Release\net5.0\UnoPhotoViewer.Wasm.dll
1>Done building project "UnoPhotoViewer.Wasm.csproj".
========== Build: 1 succeeded, 0 failed, 0 up-to-date, 0 skipped ==========
========== Deploy: 0 succeeded, 0 failed, 0 skipped ==========
Deployment output appears the same:
IsEmulatorSupported: START
	Is64BitOperatingSystem = True
DetectSlat: START
	isHyperVRunning = True
DetectSlat: END (Success, 0:00:00.21)
	SlatState = SlatAndHWVEnabled
	IsEmulatorSupported = True
IsEmulatorSupported: END (Success, 0:00:00.256)

**Reply by jeromelaban** [reply]
Thanks. When you say it does not start, what does that mean? Is it that the browser does not open? Is it that the app stays on the splash screen? Something else?

**Reply by Robotuner** [reply]
The Microsoft Visual Studio Debug Console appears. Its content is:
Hosting environment: Development
Content root path: C:\Code\Photo\UnoPhotoViewer\UnoPhotoViewer.Wasm
Now listening on: https://localhost:5001
Now listening on: http://localhost:5000
Application started. Press Ctrl+C to shut down.

C:\Program Files\dotnet\dotnet.exe (process 26280) exited with code -1.
To automatically close the console when debugging stops, enable Tools->Options->Debugging->Automatically close the console when debugging stops.
Press any key to close this window . . .

A Browser page appears, it is titled about:blank and there in nothing displayed. Looking at the developer tools/elements the body is empty:<HTML><head><head><body></body></html>

and I get the error  message popup dialog.  The wasm app never appears.

**Reply by jeromelaban** [reply]
If you navigate to the http://localhost:5000/ uri, does the app start?

**Reply by Robotuner** [reply]
hmm,  had not thought of that.  When I replace the browser Page title of about:blank with localhost:5000, the app appears!  Thanks for the help.  Is there something amiss in the configuration setting somewhere?

**Reply by jeromelaban** [reply]
Good, then this rules out any build error. This could mean that there's something wrong happening with visual studio somehow.

A few things to try:
- In the debug toolbar where you see `UnoPhotoViewer.Wasm`, do you see an option to select "IIS" ? If not, let's try something else
- Try creating a blazor webassembly application from the visual studio template, and see if the app deploys properly.

**Reply by Robotuner** [reply]
A new blazer webassembly application does deploy properly.   The Browser page initially appears with a link of about:Blank momentarily, then changes to https://localhost:7136

**Reply by jeromelaban** [reply]
Thank. When you start the Uno project without the debugger (Ctrl+F5) does it stay blank as well ?

**Reply by Robotuner** [reply]
yes, the behavior is the same regardless of whether the solution configuration is set to Debug or Release.  Checking other settings, I note that once the app is built, it will start correctly if I press the Start without Debugging (Ctrl + F5) button.  

I can live with that.
